libmscore/score.cpp

```diff
@@ -2067,59 +2034,132 @@ bool Score::appendScore(Score* score, bool addPageBreak, bool addSectionBreak)
-      for (auto sp : score->spanner()) {
-            Spanner* spanner = sp.second;
+      // clone the spanners (only in the range currently copied)
+      auto ospans = score->spanner();
+      auto lb = ospans.lower_bound(startTick.ticks()), ub = ospans.upper_bound(endTick.ticks());
+      for (auto sp = lb; sp != ub; sp++) {
+            Spanner* spanner = sp->second;
+            
+            if (spanner->tick2() > endTick) continue; // map is by tick() so this can still happen in theory...
+
             Spanner* ns = toSpanner(spanner->clone());
             ns->setScore(this);
             ns->setParent(0);
-            ns->setTick(spanner->tick() + tickOfAppend);
-            ns->setTick2(spanner->tick2() + tickOfAppend);
-            if (ns->type() == ElementType::SLUR) {
-                  // set start/end element for slur
-                  ns->setStartElement(0);
-                  ns->setEndElement(0);
-                  Measure* sm = tick2measure(ns->tick());
-                  if (sm)
-                        ns->setStartElement(sm->findChordRest(ns->tick(), ns->track()));
-                  Measure * em = tick2measure(ns->tick2());
-                  if (em)
-                        ns->setEndElement(em->findChordRest(ns->tick2(), ns->track2()));
-                  if (!ns->startElement())
-                        qDebug("clone Slur: no start element");
-                  if (!ns->endElement())
-                        qDebug("clone Slur: no end element");
-                  }
+            ns->setTick(spanner->tick() - startTick + tickOfAppend);
+            ns->setTick2(spanner->tick2() - startTick + tickOfAppend);
+            ns->computeStartElement();
+            ns->computeEndElement();
             addElement(ns);
             }
+
+      return true;
+      }
+
+//---------------------------------------------------------
+//   appendScore
+//---------------------------------------------------------
+
+bool Score::appendScore(Score* score, bool addPageBreak, bool addSectionBreak)
+      {
+      if (parts().size() < score->parts().size() || staves().size() < score->staves().size()) {
+            qDebug("Score to append has %d parts and %d staves, but this score only has %d parts and %d staves.", score->parts().size(), score->staves().size(), parts().size(), staves().size());
+            return false;
+            }
+
+      if (!last()) {
+            qDebug("This score doesn't have any MeasureBase objects.");
+            return false;
+            }
+
+      // apply Page/Section Breaks if desired
+      if (addPageBreak) {
+            if (!last()->pageBreak()) {
+                  last()->undoSetBreak(false, LayoutBreak::Type::LINE); // remove line break if exists
+                  last()->undoSetBreak(true, LayoutBreak::Type::PAGE);  // apply page break
+                  }
+            }
+      else if (!last()->lineBreak() && !last()->pageBreak())
+            last()->undoSetBreak(true, LayoutBreak::Type::LINE);
```

Comment:
Needs brackets around for the else if statement because the if statement is multilined